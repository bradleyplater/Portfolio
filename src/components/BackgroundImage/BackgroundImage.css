.image {
  height: 100%;
  width: 101%;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}
