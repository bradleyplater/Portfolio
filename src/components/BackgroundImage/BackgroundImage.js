import React from "react";
import "./BackgroundImage.css";

export default function BackgroundImage({ src }) {
  return (
    <div
      className="image"
      data-testid="background-image"
      style={{ backgroundImage: src }}
    ></div>
  );
}
