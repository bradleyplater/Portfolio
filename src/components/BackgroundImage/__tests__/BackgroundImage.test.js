import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import BackgroundImage from "../BackgroundImage";
import ComputerImage from "../Coffee.jpg";

it("Renders without Crashing", () => {
  render(<BackgroundImage src={`url(${ComputerImage})`}></BackgroundImage>);
});

it("Renders with correct styling", () => {
  const { getByTestId } = render(
    <BackgroundImage src={`url(${ComputerImage})`}></BackgroundImage>
  );
  expect(getByTestId("background-image").className).toBe("image");
  expect(getByTestId("background-image")).toHaveStyle(
    `background-image: url(${ComputerImage})`
  );
});

it("Is correct compared to snapshot", () => {
  const tree = renderer
    .create(<BackgroundImage src={`url(${ComputerImage})`}></BackgroundImage>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
