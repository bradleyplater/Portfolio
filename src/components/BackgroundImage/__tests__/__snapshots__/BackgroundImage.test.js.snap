// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is correct compared to snapshot 1`] = `
<div
  className="image"
  data-testid="background-image"
  style={
    Object {
      "backgroundImage": "url(Coffee.jpg)",
    }
  }
/>
`;
